combustion/gnns/inferer.py

```diff
@@ -0,0 +1,240 @@
+"""This module proposes classes for inference in the gnn use-case"""
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle
+import logging
+import h5py
+import networkx as nx
+import numpy
+import torch
+import torch_geometric as pyg
+from pytorch_lightning import LightningModule
+from pytorch_lightning.utilities.cli import LightningCLI
+
+CWD = os.getcwd()
+
+
+class Inferer:
```

Comment:
This mother class could be in a higher-level dir? Or we keep it here but remind to move it and reuse it when another use-case/pipeline needs it?

Reply:
Yes it could, the idea here was to show what could be envisionned in term of factorization of code in the future, but while that is not a fully shared vision, I propose to keep it here as a reminder of what could be done.